rafs/src/metadata/direct_v6.rs

```diff
@@ -1264,14 +1204,83 @@ impl RafsInode for OndiskInodeWrapper {
         self.get_child_count()
     }
 
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+}
+
+impl RafsInodeExt for OndiskInodeWrapper {
+    fn as_inode(&self) -> &dyn RafsInode {
+        self
+    }
+
+    /// Get inode number of the parent directory.
+    fn parent(&self) -> u64 {
+        match self.parent_inode.get() {
+            Some(parent) => parent,
+            None => {
+                debug_assert!(self.is_dir());
+                let ino = self.get_child_by_name(OsStr::new("..")).unwrap().ino();
+                self.parent_inode.set(Some(ino));
+                ino
+            }
+        }
+    }
+
+    /// Get name of the inode.
+    ///
+    /// # Safety
+    /// It depends on Self::validate() to ensure valid memory layout.
+    fn name(&self) -> OsString {
+        let mut curr_name = OsString::from("");
+        match self.name.borrow().as_ref() {
+            Some(name) => return name.clone(),
+            None => {
+                debug_assert!(self.is_dir());
```

Comment:
Do we still depend on `debug_assert`? This will cause different behavior between debug and release binary. Maybe use `Result<OsString> here`.

Reply:
then let's use "assert"?

Reply:
Fixed